lib/commands/deploy.js

```diff
@@ -5,18 +5,24 @@ module.exports = {
 
   availableOptions: [
     { name: 'environment', type: String, default: 'development', aliases: ['e',{'dev' : 'development'}, {'prod' : 'production'}] },
-    { name: 'deploy-config-file', type: String, default: 'config/deploy.js' }
+    { name: 'deploy-config-file', type: String, default: 'config/deploy.js' },
+    { name: 'activate', type: Boolean, default: false }
   ],
 
   run: function(commandOptions, rawArgs) {
     process.env.DEPLOY_ENVIRONMENT = commandOptions.environment;
 
+    var commandLineArgs = {
+      activateRevision: commandOptions.activate
+    };
+
     var PipelineTask = require('../tasks/pipeline');
     var pipeline = new PipelineTask({
       project: this.project,
       ui: this.ui,
       deployEnvironment: commandOptions.environment,
       deployConfigPath: commandOptions.deployConfigFile,
+      commandLineArgs: commandLineArgs,
       hooks: [
```

Comment:
Not sure how this would work exactly, but what do you think about making these hooks include "willActivate", "activate", "didActivate" only if immediate activation is enabled?